src/events/IonEventStream.ts

```diff
@@ -146,49 +183,61 @@ export class IonEventStream {
-                            this.endContainer(currentContainer.pop()!, currentContainerIndex.pop()!);
+        return false;
+    }
+
+    private generateStream(): void {
+        try{
+            let tid: IonType | null = this.reader.next();
+            if (tid === IonTypes.SYMBOL && this.reader.stringValue() === "$ion_event_stream") {
+                this.marshalStream();
+                this.isEventStream = true;
+                return;
+            }
+            let currentContainer: IonEvent[] = [];
+            let currentContainerIndex: number[] = [];
+            while (true) {
+                if (this.reader.isNull()) {
+                    this.eventStream.push(this.eventFactory.makeEvent(IonEventType.SCALAR, tid!, this.reader.fieldName(), this.reader.depth(), this.reader.annotations(), true, this.reader.value()));
+                } else {
+                    switch (tid) {
+                        case IonTypes.LIST:
+                        case IonTypes.SEXP:
+                        case IonTypes.STRUCT: {
+                            let containerEvent = this.eventFactory.makeEvent(IonEventType.CONTAINER_START, tid, this.reader.fieldName(), this.reader.depth(), this.reader.annotations(), false, null);
+                            this.eventStream.push(containerEvent);
+                            currentContainer.push(containerEvent);
+                            currentContainerIndex.push(this.eventStream.length);
+                            this.reader.stepIn();
+                            break;
+                        }
+                        case null: {
+                            if (this.reader.depth() === 0) {
+                                this.eventStream.push(this.eventFactory.makeEvent(IonEventType.STREAM_END, IonTypes.NULL, null, this.reader.depth(), [], false, undefined));
+                                return;
+                            } else {
+                                this.reader.stepOut();
+                                this.endContainer(currentContainer.pop()!, currentContainerIndex.pop()!);
+                            }
+                            break;
+                        }
+                        default: {
+                            this.eventStream.push(this.eventFactory.makeEvent(IonEventType.SCALAR, tid, this.reader.fieldName(), this.reader.depth(), this.reader.annotations(), false, this.reader.value()));
+                            break;
                         }
-                        break;
-                    }
-                    default: {
-                        this.eventStream.push(this.eventFactory.makeEvent(IonEventType.SCALAR, tid, this.reader.fieldName(), this.reader.depth(), this.reader.annotations(), false, this.reader.value()));
-                        break;
                     }
                 }
+                tid = this.reader.next();
             }
-            tid = this.reader.next();
+        } catch (er) {
```

Comment:
Nit: `er` isn't a commonly used abbreviation for "error." How about `err` or just `error`?